source/server/filter_chain_manager_impl.cc

```diff
@@ -496,6 +562,23 @@ FilterChainManagerImpl::findFilterChain(const Network::ConnectionSocket& socket)
              : default_filter_chain_.get();
 }
 
+const Network::FilterChain*
+FilterChainManagerImpl::findFilterChainUsingMatcher(const Network::ConnectionSocket& socket) const {
+  Network::Matching::MatchingDataImpl data(socket);
+  const auto match_result = Matcher::evaluateMatch<Network::MatchingData>(*matcher_, data);
+  ASSERT(match_result.match_state_ == Matcher::MatchState::MatchComplete,
+         "Matching must complete for network streams.");
+  if (match_result.result_) {
+    const auto result = match_result.result_();
+    const auto chain_match =
+        filter_chains_by_name_.find(result->getTyped<FilterChainNameAction>().name_);
```

Comment:
Since filter_chains_by_name_ is immutable after the listener build,
I am thinking about the result contains the reference to the filter chain, rather than the string name.

It might be an early optimization.

Reply:
Good idea. Let me try add a context for the actions with the mapping.